autowarefoundation/autoware
# Feature/pacmod apex as timer publisher

## Status
** DEVELOPMENT**

## Description
This change enables publishing at a constant cycle regardless of the update of input topics.
For example, even when the update of the self-position estimate is delayed, the control command to the row-level controller is updated at the specified cycle.

## Steps to Test or Reproduce
1. Launch this node with `use_timer_publisher` option.

```
$ roslaunch as as_interface.launch use_timer_publisher:=true
```

1. Check topics

```
$ rostopic hz /as/arbitrated_speed_commands
subscribed to [/as/arbitrated_speed_commands]
average rate: 10.001
	min: 0.096s max: 0.105s std dev: 0.00234s window: 10
average rate: 10.007
	min: 0.096s max: 0.105s std dev: 0.00231s window: 20
average rate: 10.005
	min: 0.096s max: 0.105s std dev: 0.00225s window: 30
```
